tests/parser_test.py

```diff
@@ -427,3 +428,98 @@ def test_keyword_identifiers(text: str, rule: str) -> None:
     """
     unit = parse_buffer(text, rule=rule)
     assert len(unit.diagnostics) == 0, text + "\n".join(str(d) for d in unit.diagnostics)
+
+
+@pytest.mark.parametrize(
+    "string,expected",
+    [
+        (
+            "(case A is when V1 => 1, when V2 => 2)",
+            {
+                "_kind": "CaseExpression",
+                "choices": [
+                    {
+                        "_kind": "Choice",
+                        "expression": {"_kind": "NumericLiteral", "_value": "1"},
+                        "selector": {
+                            "_kind": "ID",
+                            "name": {"_kind": "UnqualifiedID", "_value": "V1"},
+                            "package": None,
+                        },
+                    },
+                    {
+                        "_kind": "Choice",
+                        "expression": {"_kind": "NumericLiteral", "_value": "2"},
+                        "selector": {
+                            "_kind": "ID",
+                            "name": {"_kind": "UnqualifiedID", "_value": "V2"},
+                            "package": None,
+                        },
+                    },
+                ],
+                "expression": {
+                    "_kind": "Variable",
+                    "identifier": {
+                        "_kind": "ID",
+                        "name": {"_kind": "UnqualifiedID", "_value": "A"},
+                        "package": None,
+                    },
+                },
+            },
+        ),
+        (
+            "M'(Len => (case L is when L1 => 8, when L2 => 16))",
+            {
+                "_kind": "MessageAggregate",
```

Comment:
I think it can be confusing that parsing of a message aggregate is tested inside `test_extended_case_expression`. If the syntax of message aggregates are changed in some way, `test_extended_case_expression` would fail.

Reply:
That's true. Removed.